### PR TITLE
Fixes #2240: Audit of microservices to use fetch from Satellite

### LINKED ISSUE #2240: Use Satellite's fetch() function and remove local deps for microservices
Now that #2135 is fixed, and v1.16.0 of Satellite is released, we can use `const { fetch }  = require('@senecacdot/satellite')` to get a `fetch()` function in our microservices.  Any of the services that are currently using `fetch()` or `got()` or the like to do client HTTP requests should switch to use this.

### PR DESCRIPTION
**Pull request by HyperTHD**
## Issue This PR Addresses
This PR closes #2240 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

Satellite version 1.16 has added node-fetch to it's api. This PR does an audit of all microservices that use `node-fetch` or `got` and switches to use fetch provided by Satellite.

NOTE: I'm not quite sure how to replicate `got.stream` with fetch, so I appreciate any feedback on that. Everything else should hopefully be good.

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/api/feed-discovery/src/middleware.js [hunk 1]
```diff
@@ -20,14 +19,14 @@ module.exports.checkValidUrl = function checkValidUrl() {
 module.exports.checkValidBlog = function checkValidBlog() {
   return async (req, res, next) => {
     try {
-      const { statusCode, headers, body } = await got(req.body.blogUrl);
-      const contentType = headers['content-type'];
+      const response = await fetch(req.body.blogUrl);
+      const contentType = response.headers;
```
Comment (humphd) [inline src/api/feed-discovery/src/middleware.js:23]:
```js
const contentType = response.headers.get('content-type);
```

File: src/api/image/bin/unsplash-download.js [hunk 2]
```diff
@@ -18,7 +17,7 @@ const pipeline = promisify(stream.pipeline);
  * @param {string} filename - the local filename to use when writing
  */
 function downloadPhoto(url, filename) {
-  return pipeline(got.stream(url), fs.createWriteStream(filename)).then(() =>
+  return pipeline(fetch(url), fs.createWriteStream(filename)).then(() =>
```
Comment (humphd) [inline src/api/image/bin/unsplash-download.js:20]: I think this has to be the `body` of the response to get a stream, see https://github.com/node-fetch/node-fetch#streams

**Review by humphd — CHANGES_REQUESTED** [review]
Nice work, a few things with the switch over from `got`.

File: src/api/image/bin/unsplash-download.js (hunk 2 above)
Comment (HyperTHD) [reply src/api/image/bin/unsplash-download.js]: Thanks! Will do

File: src/api/feed-discovery/src/middleware.js (hunk 1 above)
Comment (HyperTHD) [reply src/api/feed-discovery/src/middleware.js]: Thanks!

File: src/api/image/bin/unsplash-download.js [hunk 3]
```diff
@@ -16,8 +16,9 @@ const pipeline = promisify(stream.pipeline);
  * @param {string} url - the URL to the image
  * @param {string} filename - the local filename to use when writing
  */
-function downloadPhoto(url, filename) {
-  return pipeline(fetch(url), fs.createWriteStream(filename)).then(() =>
+async function downloadPhoto(url, filename) {
+  const response = await fetch(url);
```
Comment (humphd) [inline src/api/image/bin/unsplash-download.js:20]: I think `await` is incorrect here.  Node has a number of programming models: synchronous, asynchronous, and streams.  In this code, you are trying to create a pipeline out of two streams, where the `fetch` stream gets piped into the file write stream.  That way, as the bytes flow in from the `fetch`, they will get written to disk without having to accumulate in memory.

**Review by humphd — CHANGES_REQUESTED** [review]
Looks like this is failing tests, so I suspect he switch from `got` needs more work.  Confirm we have the API usage correct with `node-fetch`.

**Comment by manekenpix** [comment]
@HyperTHD what's the progress on this? can we include it in the 2.1 release?

**Comment by HyperTHD** [comment]
> @HyperTHD what's the progress on this? can we include it in the 2.1 release?

ik 2.1 was just released but still gonna comment regardless

The main issue here is that tests need to be updated as well. Not to mention that `got` has uses here that aren't used with `node-fetch`. I'm gonna take another look through the files when I get some time and put up a commit that should hopefully get these tests to pass.

**Pull request closed without merging**